build_rules.gradle

```diff
@@ -421,7 +421,7 @@ ext.applyJavaNature = {
     if (configuration.enableErrorProne) {
       options.compilerArgs += [
         "-XepDisableWarningsInGeneratedCode",
-        "-XepExcludedPaths:.*/build/generated.*avro-java/.*",
+        "-XepExcludedPaths:(.*/)?(build/generated.*avro-java|build/generated)/.*",
```

Comment:
Out of curiosity, does the build fail without this additional path? My understanding was that [`-XepDisableWarningsInGeneratedCode`](http://errorprone.info/docs/flags) should disable generated code warnings, but we needed the additional exclusion because Avro doesn't add the `@javax.annotation.Generated` annotation.

Reply:
yeah the generated class, `beam/sdks/java/extensions/protobuf/build/generated/source/proto/test/java/org/apache/beam/sdk/extensions/protobuf/Proto2CoderTestMessages.java` has warnings for `MissingOverride` and `ReferenceEquality`

Reply:
Cool, thanks for the explanation. With a bit of digging I found an open PR to [add `@Generated` annotations to protobuf](https://github.com/google/protobuf/issues/42). Apparently there is an undocumented command line parameter, but I don't think we should rely on it. I agree with the route you've taken.